reporters/plain_reporter.py

```diff
@@ -12,8 +13,52 @@ def handle_message(self, msg):
 
     def print_message_ids(self):
         # Sort the messages by their type.
-        self._messages.sort(key=lambda s: s[0])
-
+        #self._messages.sort(key=lambda s: s[0])
+        self.sort_messages()
         for msg in self._messages:
             code = msg.msg_id
             print(code, '({})\n    {}'.format(msg.symbol, msg.msg))
+
+    def sort_messages(self):
+        # Sort the messages by their type.
+        self._messages.sort(key=lambda s: s[0])
+
+        i = 0
+        while i < len(self._messages):
+            current_id = self._messages[i].msg_id
+            count = 1
+            messages = ''
+            msg_count = 1
+            while i+1 < len(self._messages) and self._messages[i+1].msg_id == current_id:
+                count += 1
+                if msg_count <= 5:
+                    messages += '\n    ' + self._messages[i+1].msg
```

Comment:
Make `messages` a list of strings rather than a string; this allows you to do a `len` to make the "max 5" check. You can call `'\n    '.join(messages)` later on.